bokehjs/src/lib/api/palettes.ts

```diff
@@ -1011,6 +1012,43 @@ export const colorblind = {
   Colorblind,
 }
 
+export function interp_palette(palette: Color[], n: number): Color[] {
+  const npalette = palette.length
+  if (npalette < 1)
+    throw new Error("palette must contain at least one color")
+  if (n < 0)
+    throw new Error("requested palette length cannot be negative")
+
+  // Arrayable.interpolate operates on whole arrays, not slices, so need separate
+  // arrays for each of the R, G, B and A components.
+  const r = new Uint8Array(n)
+  const g = new Uint8Array(n)
+  const b = new Uint8Array(n)
+  const a = new Uint8Array(n)
+  for (let i = 0; i < npalette; i++) {
+    const rgba = color2rgba(palette[i])
```

Comment:
Would be better to unpack here (`const [r, g, b, a] = ...`) and avoid indexing later.